components/cam/src/dynamics/se/stepon.F90

```diff
@@ -128,26 +133,36 @@ subroutine stepon_init(dyn_in, dyn_out )
   call addfld ('DIFFU   ', (/ 'ilev' /),'A', 'm/s2    ','U horizontal diffusion',                      gridname='physgrid')
   call addfld ('DIFFV   ', (/ 'ilev' /),'A', 'm/s2    ','V horizontal diffusion',                      gridname='physgrid')
   call register_vector_field('DIFFU', 'DIFFV')
-  
   call addfld ('ETADOT', (/ 'ilev' /), 'A', '1/s', 'Vertical (eta) velocity', gridname='physgrid')
-  call addfld ('U&IC',   (/ 'lev' /),  'I', 'm/s', 'Zonal wind',              gridname='physgrid' )
-  call addfld ('V&IC',   (/ 'lev' /),  'I', 'm/s', 'Meridional wind',         gridname='physgrid' )
-  ! Don't need to register U&IC V&IC since we don't interpolate IC files
-  call add_default ('U&IC',0, 'I')
-  call add_default ('V&IC',0, 'I')
 
-  call addfld ('PS&IC', horiz_only,  'I', 'Pa', 'Surface pressure',gridname='physgrid')
-  call addfld ('T&IC',  (/ 'lev' /), 'I', 'K',  'Temperature',     gridname='physgrid')
+  if (fv_nphys > 0) then
+    grid_name = 'GLL'
+  else
+    grid_name = 'physgrid'
+  end if 
 
-  call add_default ('PS&IC      ',0, 'I')
-  call add_default ('T&IC       ',0, 'I')
+  call addfld ('PS&IC',horiz_only,'I','Pa', 'Surface pressure',gridname=grid_name)
+  call addfld ('U&IC', (/'lev'/), 'I','m/s','Zonal wind',      gridname=grid_name)
+  call addfld ('V&IC', (/'lev'/), 'I','m/s','Meridional wind', gridname=grid_name)
+  call addfld ('T&IC', (/'lev'/), 'I','K',  'Temperature',     gridname=grid_name)
   do m = 1,pcnst
-     call addfld (trim(cnst_name(m))//'&IC', (/ 'lev' /), 'I', 'kg/kg', cnst_longname(m), gridname='physgrid')
+    call addfld (trim(cnst_name(m))//'&IC',(/'lev'/),'I','kg/kg',cnst_longname(m),gridname=grid_name)
   end do
+  
+  call add_default ('U&IC',0, 'I')
+  call add_default ('V&IC',0, 'I')
+  call add_default ('PS&IC      ',0, 'I')
+  call add_default ('T&IC       ',0, 'I')
   do m = 1,pcnst
-     call add_default(trim(cnst_name(m))//'&IC',0, 'I')
+    call add_default(trim(cnst_name(m))//'&IC',0, 'I')
   end do
 
+  call addfld('DYN_T'    ,(/ 'lev' /), 'A', 'K',    'Temperature (dyn grid)', gridname='GLL')
```

Comment:
I might be missing something, but these look like new addfld calls that run even in the baseline code path. Is this intended?

Reply:
That's correct. I can wrap them in an if statement if it's better to hide them.
When I initially added there I had a misconception that using the "GLL" grid would output ALL the GLL nodes, including the redundant information on the edge. So the DYN_U and DYN_V might be duplicates of the other GLL output variables. I don't think there's any performance implications to having extra addfld calls, but I can remove these. They are beneficial for the physgrid runs for verfiying that the phys and dyn states are in sync.

Reply:
I did not pay enough attention to this before: what does addfld do? What did you mean by "They are beneficial for the physgrid runs for verfiying that the phys and dyn states are in sync."? Thanks!

Reply:
addfld just adds these variables to the list of variables that are available for output. The outfld calls don't do anything if the namelist doesn't request that specific variable. 

I wanted to have certain variables on both grids to verify that the mapping was working as expected. I later realized that certain comparison were nonsense, like since physics cannot change surface pressure, the DYN_PS and PS fields will be essentially identical. However, if someone else wants to be able to look at a field on both grids simultaneously in the same output file then they can mimic the calls for these variables.